tests/detabler/detabler.js

```diff
@@ -0,0 +1,187 @@
+			var editor = this.editorBot.editor,
+				startHtml,
+				endHtml;
+
+			startHtml = '<p>foobar^</p>' +
+				'<p>baz</p>' +
+				'<table>' +
+					'<tbody>' +
+						'<tr><td>leave me alone</td></tr>' +
+					'</tbody>' +
+				'</table>';
+
+			endHtml = '<p>foobar</p>' +
+				'<p>baz</p>' +
+				'<table>' +
+					'<tbody>' +
+						'<tr><td>leave me alone</td></tr>' +
+					'</tbody>' +
+				'</table>';
+
+			this.editorBot.setHtmlWithSelection(startHtml);
+			// setHtmlWithSelection doesn't appear to refresh command state, so we must manually do it
+			this.command.refresh(editor, editor.elementPath());
+
+			assert.areEqual(CKEDITOR.TRISTATE_OFF, this.command.state);
+
+			editor.execCommand('detable');
+
+			this.assertHtml(endHtml, editor.getData(), 'Editor data does not match.');
+		},
+		'it converts table cells to p tags': function () {
+			var editor = this.editorBot.editor,
+				startHtml,
+				startHtmlWithoutSelection,
+				endHtml;
+
+			startHtml = '<p>foo</p>' +
+				'<table>' +
+					'<tbody>' +
+						'<tr><td>^foo</td><td>bar</td></tr>' +
+					'</tbody>' +
+				'</table>';
+			startHtmlWithoutSelection = '<p>foo</p><table><tbody><tr><td>foo</td><td>bar</td></tr></tbody></table>';
+			endHtml = '<p>foo</p><p>foo</p><p>bar</p>';
+
+			this.editorBot.setHtmlWithSelection(startHtml);
+			// setHtmlWithSelection doesn't appear to refresh command state, so we must manually do it
+			this.command.refresh(editor, editor.elementPath());
+
+			assert.areEqual(CKEDITOR.TRISTATE_ON, this.command.state);
+
+			editor.execCommand('detable');
+
+			this.assertHtml(endHtml, editor.getData(), 'Editor data does not match.');
+
+			editor.execCommand('undo');
+
+			this.assertHtml(startHtmlWithoutSelection, editor.getData(), 'Editor data does not match after being undone.');
+		},
+		'it does not create nested p': function () {
+			var editor = this.editorBot.editor,
+				startHtml,
+				endHtml;
+
+			startHtml = '<table>' +
+					'<tbody>' +
+						'<tr><td><p>^foo</p></td><td>bar</td></tr>' +
+					'</tbody>' +
+				'</table>';
+			endHtml = '<p>foo</p><p>bar</p>'; // the p tag for bar gets created because you can't have a parentless text node
+
+			this.editorBot.setHtmlWithSelection(startHtml);
+			// setHtmlWithSelection doesn't appear to refresh command state, so we must manually do it
+			this.command.refresh(editor, editor.elementPath());
+
+			assert.areEqual(CKEDITOR.TRISTATE_ON, this.command.state);
+
+			editor.execCommand('detable');
+
+			this.assertHtml(endHtml, editor.getData(), 'Editor data does not match.');
+		},
+		'it handles multiple non-text children in the same cell': function () {
+			var editor = this.editorBot.editor,
+				startHtml,
+				endHtml;
+
+			startHtml = '<table>' +
+					'<tbody>' +
+						'<tr><td><p>^foo</p><p>bar</p></td></tr>' +
+				'</tbody>' +
+			'</table>';
+			endHtml = '<p>foo</p><p>bar</p>'; // the p tag for bar gets created because you can't have a parentless text node
```

Comment:
Good catch.  

It used to actually apply, now it doesn't actually apply to anything, because of a side effect of fixing the multiple text nodes case, I think.